shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -281,6 +287,18 @@ bool FlutterWindowsEngine::RunWithEntrypoint(const char* entrypoint) {
   return true;
 }
 
+void FlutterWindowsEngine::OnVsync(intptr_t baton) {
+  auto current_time = embedder_api_.GetCurrentTime();
```

Comment:
Is current time in nanoseconds? Because `frame_start_time_nanos` expects it to be in nanoseconds. If not, the times will be in the past and the engine will keep scheduling frames as fast as possible. I suggest using std::chrono everywhere as it is easy to mess up units.